### PR TITLE
refactor: maxStreak calculation;

### PR DESCRIPTION
**Pull request by ilasw**
test: add test for maxStreak calc;

### CODE REVIEW

**Review by sshanzel — COMMENTED** [review]
Also, in the mutation, we should also clear the cache after successful update. Along with the alerts proeprty being changed.

File: __tests__/users.ts
Comment (sshanzel) [inline __tests__/users.ts:820]: Instead of deleting the redis key, we should check here if it was deleted.

**Review by sshanzel — APPROVED** [review]
Tests looked amazing. Just update the last part and wait for Ante's approval before merging 🚀

**Review by capJavert — APPROVED** [review]

**Review by nensidosari — APPROVED** [review]

**Pull request merged**
